diff --git a/themes/book/app/routes/$.tsx b/themes/book/app/routes/$.tsx
index 9639893c..6a2fb591 100644
--- a/themes/book/app/routes/$.tsx
+++ b/themes/book/app/routes/$.tsx
@@ -1,3 +1,4 @@
+import { useMemo } from 'react';
 import {
   json,
   type V2_MetaFunction,
@@ -30,6 +31,7 @@ import { MadeWithMyst } from '@myst-theme/icons';
 import { ComputeOptionsProvider, ThebeLoaderAndServer } from '@myst-theme/jupyter';
 import { ArticlePage } from '../components/ArticlePage.js';
 import type { TemplateOptions } from '../types.js';
+import { selectAll } from 'unist-util-select';
 import { useRouteError, isRouteErrorResponse } from '@remix-run/react';
 type ManifestProject = Required<SiteManifest>['projects'][0];
 
@@ -149,6 +151,14 @@ export default function Page() {
     ...siteDesign,
     ...pageDesign,
   };
+  const _ = useMemo(() => {
+    const mdast = data.page.mdast;
+    selectAll('block', mdast).forEach((node) => {
+      if ('class' in ((node as any).data ?? {})) {
+        (node as any).class = (node.data as any).class;
+      }
+    });
+  }, [data.page.mdast]);
   return (
     <ArticlePageAndNavigation
       hide_toc={hide_toc}
